### DISCUSSION TITLE
lns focus on

### DISCUSSION THREAD

**Discussion by C8504** [google/or-tools#2771] [CP-SAT questions] [answered]
Hi experts,

Could I have a question about set LNS fixed variables and relaxed variables？

For example, suppose we have X and Y variables in the model, could we specify X as fixed and Y as relaxed by sat parameter or mark them by modeling api?

Thanks.
John

**Comment by lperron** [comment, marked as answer]
No. The LNS code is internal.

You can always to 'external' LNS by reducing the domains of the fixed variables, and solving the new model using the CP-SAT solver.

**Reply by C8504** [reply]
Got it, thanks.
